pkg/kubelet/container/helpers.go

```diff
@@ -134,6 +135,18 @@ func HashContainerWithoutResources(container *v1.Container) uint64 {
 	return uint64(hashWithoutResources.Sum32())
 }
 
+// HashAuth - returns a hash code for a CRI pull image auth
+func HashAuth(auth *runtimeapi.AuthConfig) string {
+	hash := fnv.New64a()
+	authJSON, err := json.Marshal(auth)
+	if err != nil {
+		klog.ErrorS(err, "Failed to marshal auth config")
+		return ""
```

Comment:
Why to suppress the error and return empty string? Why not return the (string, error)?

I see that this has been used in https://github.com/kubernetes/kubernetes/pull/114847/files#diff-b22ab169cdc70b2e968bb653f29cf0a1384d3bccaebeef96aebf2a5a40821bbeR367. But there is an already comment by @rphillips on that part of the code to return the error.

Reply:
Fixed.